### ISSUE TITLE
Need to display training progress on the GUI interface

### ISSUE THREAD

**Issue by monkeycc** [ultralytics/ultralytics#3554]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

```python
from ultralytics import YOLO

# Load a model
model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)

# Train the model
model.train(data='coco128.yaml', epochs=100, imgsz=1280)
```

Need to display training progress on the GUI interface

How should real-time training log information be called?

```
      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
     1/1000      4.68G      1.775      3.227      14.09      1.145          9       1280: 100%|██████████| 20/20 [00:04
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP
                   all         19         75        0.5    0.00746   9.44e-05   5.62e-05      0.501    0.00746    0.00108   0.000508

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
     2/1000      4.85G      1.728      2.184      14.13      1.012         19       1280:  70%|███████   | 14/20 [00:02
```

ultralytics/yolo/v8/detect/train.py
```python
def progress_string(self):
    """Returns a formatted string of training progress with epoch, GPU memory, loss, instances and size."""
    return ('\n' + '%11s' *
            (4 + len(self.loss_names))) % ('Epoch', 'GPU_mem', *self.loss_names, 'Instances', 'Size')
```
I don't know how to call the data here

**Comment by glenn-jocher** [comment]
@monkeycc hi there,

Thank you for reaching out to us. Currently, displaying training progress on the GUI interface is not supported in YOLOv8. The training progress is displayed in the command line interface (CLI) using a formatted string that includes information such as epoch, GPU memory, loss values, instances, and size.

In the example code you provided, the `progress_string` function in the `train.py` file generates the formatted string. However, this string is printed to the command line, not the GUI interface.

If you would like to display the training progress on the GUI interface, you would need to implement the necessary code and user interface components to capture and visualize the progress in your desired way. The specific implementation details would depend on the GUI framework or library you are using.

Please let me know if you have any further questions or need additional assistance.

Best regards,
Glenn Jocher

**Comment by glenn-jocher** [comment]
@monkeycc you can try Ultralytics HUB for training visuals. It's free :)

- HUB: https://hub.ultralytics.com/
- YOLOv8s example page: https://hub.ultralytics.com/models/4iiEIb7peyXxmzwkUbVN

<img width="1481" alt="Screenshot 2023-07-07 at 23 38 37" src="https://github.com/ultralytics/ultralytics/assets/26833433/e6640cb7-ec76-456a-906e-3394961404cc">
